### ISSUE TITLE
Unable to update core to D10 

### ISSUE THREAD

**Issue by aannie2512** [mglaman/composer-drupal-lenient#23]
I'm trying to get the rate-limits module, which does not have a D10 version yet, to work with D10.
I installed a patch that changed rate_limits.info.yml  so core_version_requirements include  version 10,  and Upgrade Status module says the rate_limits module is now compatible.
But when I try to composer update to D10, I get the error below. Not sure what I am missing. I  installed mglaman/composer-drupal-lenient and added drupal/rate_limits to the allowed-list of lenient modules. But it still gives this error.

Problem 1
- Root composer.json requires drupal/rate_limits dev-2.x -> satisfiable by drupal/rate_limits[dev-2.x].
- drupal/rate_limits dev-2.x requires drupal/core ^8 || ^9 -> found drupal/core[8.0.0-beta6, ..., 8.9.x-dev, 9.0.0-alpha1, ..., 9.5.x-dev] but it conflicts with your root composer.json require (^10).

**Comment by mglaman** [comment]
Can you copy the relevant part of your `composer.json` with the lenient config? I'm not sure why it isn't applying.

Do you have an existing `composer.lock`? If not, that can cause some weird results per #13. The fix is to add this as a global Composer plugin as well.

**Comment by aannie2512** [comment]
Thanks for getting back so promptly.
 I didnt have composer.lock.  I  had deleted   composer.lock and vendor folder and was running 'composer install'. 

Now I reverted   core to version 9 in composer.json, and then generated composer.lock and vendor folder. After that   I updated core to version 10 in composer.json, ran 'composer update', and the lenient config was picked up, and composer upgraded to D10.
Thanks for all your help. This module has been most helpful
